File: grader/status.py
import glob
import json

assignments = glob.glob('./*.json')

students = {}
for assignment in assignments:
    with open(assignment, 'r') as f:
        data = json.loads(f.read())
        for student in data:

            try:
                if student['name'] not in students:
                    students[student['name']] = {
                        'name': student['name'],
                        'incomplete': [],
                        'failing': False
                    }

                if student['submission'] == 'none':
                    students[student['name']]['incomplete'].append(int(assignment[3:4]))
                    if len(students[student['name']]['incomplete']) > 2:
                        students[student['name']]['failing'] = True
            except Exception, e:
                print student



print json.dumps(filter(lambda student: student['failing'], students.values()))

File: grader/status.js
(function() {
	'use strict';

	var data = {
		students: document.querySelectorAll('td.student'),
		states: document.querySelectorAll('td.state > .text'),
	};

	var statuses = Array.prototype.map.call(data.students, function(student, i) {
		return {
			name: student.innerText,
			submission: data.states[i].innerText.toLowerCase()
		};
	});

	var a = document.createElement('a');
	a.href = window.URL.createObjectURL(new window.Blob([ JSON.stringify(statuses, null, '  ') ], {
		type: 'application/json'
	}));
	a.download = window.prompt('Assignment Name:') + '.json';
	a.click();

})();